### PR TITLE
Improve qpid::config::* defines

### PR DESCRIPTION
**Pull request by ekohl**
See individual commits for exact changes.

### CODE REVIEW

File: manifests/config/bind.pp [hunk 1]
```diff
@@ -25,12 +25,16 @@
...
     command  => "bind ${exchange} ${queue} ${title}",
     unless   => "exchanges ${exchange} -r | grep ${title}",
     hostname => $hostname,
     port     => $port,
     ssl_cert => $ssl_cert,
     ssl_key  => $ssl_key,
   }
+
+  Qpid::Config::Exchange <| exchange == $exchange |> -> Qpid::Config_cmd[$cmd]
+  Qpid::Config::Queue <| queue == $queue |> -> Qpid::Config_cmd[$cmd]
```
Comment (ehelms) [inline manifests/config/bind.pp]: So this adds a requirement that the queue and exchange were created/managed by Puppet somewhere within the puppet apply that includes this module, if I read correctly?

File: manifests/config/queue.pp [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# Define: qpid::config::queue
+#
+# This define ensures an queue exists
```
Comment (ehelms) [inline manifests/config/queue.pp]: s/a/an

File: manifests/config/queue.pp [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+define qpid::config::queue(
+  $queue = $title,
+  $hostname = undef,
+  $port = undef,
+  $ssl_cert = undef,
+  $ssl_key = undef,
+)
+{
+  qpid::config_cmd { "ensure queue ${queue}":
+    command  => "add queue ${queue} --durable",
```
Comment (ehelms) [inline manifests/config/queue.pp]: Should durable be an option here?

File: manifests/config/bind.pp (hunk 1 above)
Comment (ekohl) [inline manifests/config/bind.pp]: Correct. It also works if they aren't created/managed (at least, there's tests for that and I don't see any errors)

File: manifests/config/queue.pp (hunk 3 above)
Comment (ekohl) [inline manifests/config/queue.pp]: It's what puppet-katello does. I also thought about parametrizing this and wondered what the purpose of adding non-durable queues using puppet is.

File: manifests/config/queue.pp (hunk 2 above)
Comment (ekohl) [inline manifests/config/queue.pp]: Result of ```s/exchange/queue```. Fixing.

**Comment by ekohl** [comment]
Rebased and s/an/a

**Pull request merged**